bitnami/minio/templates/provisioning-job.yaml

```diff
@@ -69,29 +69,57 @@ spec:
+                  mc admin policy update {{ $minioAlias }} $3 $1=$2;
+                fi;
+              };
+
               {{- $minioUrl := printf "$MINIO_SCHEME://%s:%d" (include "common.names.fullname" .) (.Values.service.ports.api | int) }}
               {{- $minioRootUser := ternary ("$(<$MINIO_ROOT_USER_FILE)") ("$MINIO_ROOT_USER") (.Values.auth.useCredentialsFiles) }}
               {{- $minioRootPassword := ternary ("$(<$MINIO_ROOT_PASSWORD_FILE)") ("$MINIO_ROOT_PASSWORD") (.Values.auth.useCredentialsFiles) }}
               mc alias set {{ $minioAlias }} {{ $minioUrl }} {{ $minioRootUser }} {{ $minioRootPassword }};
 
+              {{- range $config := .Values.provisioning.config }}
+              {{- $options := list }}
+              {{- range $name, $value := $config.options }}
+              {{- $options = (printf "%s=%s" $name $value) | append $options }}
+              {{- end }}
+              {{- $options := join " " $options }}
+              mc admin config set {{ $minioAlias }} {{ $config.name }} {{ $options }};
+              {{- end }}
+
+              mc admin service restart {{ $minioAlias }};
+
               {{- range $user := .Values.provisioning.users }}
               mc admin user add {{ $minioAlias }} {{ $user.username }} {{ $user.password }};
+              {{- if $user.setPolicies }}
+              mc admin policy set {{ $minioAlias }} "{{ join "," $user.policies }}" user={{ $user.username }};
+              {{- else }}
               {{- range $policy := $user.policies }}
-              mc admin policy update {{ $minioAlias }} {{ $policy }} user={{ $user.username }};
+              addPolicy user {{ $user.username }} {{ $policy }};
+              {{- end }}
               {{- end }}
               {{- $userStatus := ternary ("disable") ("enable") (and (not (empty $user.disabled)) $user.disabled) }}
               mc admin user {{ $userStatus }} {{ $minioAlias }} {{ $user.username }};
               {{- end }}
 
               {{- range $group := .Values.provisioning.groups }}
               mc admin group add {{ $minioAlias }} {{ $group.name }} {{ join " " $group.members }};
+              {{- if $group.setPolicies }}
+              mc admin policy set {{ $minioAlias }} "{{ join "," $group.policies }}" group={{ $group.name }};
+              {{- else }}
               {{- range $policy := $group.policies }}
-              mc admin policy update {{ $minioAlias }} {{ $policy }} group={{ $group.name }};
+              addPolicy group {{ $group.name }} {{ $policy }};
+              {{- end }}
               {{- end }}
               {{- $groupStatus := ternary ("disable") ("enable") (and (not (empty $group.disabled)) $group.disabled) }}
               mc admin group {{ $groupStatus }} {{ $minioAlias }} {{ $group.name }};
               {{- end }}
 
+              {{- $isDistributedMode := and (eq .Values.mode "distributed") (not .Values.gateway.enabled) }}
```

Comment:
Can you move this closer to where it's actually used?

Reply:
Only if I add another variable that contains `.Values`, because `.` is overwritten in `range` and `.Values` does not exist there. Or is there a trick I don't know?

Reply:
Fair point